### PR TITLE
Make noarch

### LINKED ISSUE #3: Use `noarch`
xref https://github.com/conda-forge/cuda-feedstock/issues/4

> As this package doesn't have any architecture dependency on its own, it would make sense to make this noarch. This would allow this package to be built once and reused on other architectures/OSes, which would simply the release process and cutdown on resource usage. For dependencies that are architecture/OS dependent, the right package would be pulled in. Raising to track evaluating this option

### PR DESCRIPTION
**Pull request by adibbley**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/cuda-libraries-static-feedstock/issues/3

### CODE REVIEW

**Comment by adibbley** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,11 +9,12 @@ source:
...
 
 build:
-  number: 1
-  skip: true  # [osx or win]
+  number: 2
+  noarch: generic
 
 requirements:
   run:
+    - __linux
```
Comment (jakirkham) [inline recipe/meta.yaml]: This should ensure that the constructed `noarch` packages are generate for the intended OS & architecture (and are only installable there)

```suggestion
    - __linux               # [linux]
    - __win                 # [win]
    - __archspec * x86_64   # [x86_64]
    - __archspec * aarch64  # [aarch64]
    - __archspec * ppc64le  # [ppc64le]
```

**Review by jakirkham — COMMENTED** [review]
Thanks Alex! 🙏

Since this has OS specific requirements, think we need to take a slightly different approach.

When adding `noarch` to packages with OS specific requirements, would follow [this conda-forge doc]( https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-packages-with-os-specific-dependencies )

Would add this to `conda-forge.yml`:

```yaml
noarch_platforms:
  - linux_64
  - linux_aarch64
  - linux_ppc64le
  - win_64
```

Also would make the changes below to `meta.yaml`

Think we will need a re-render afterwards

**Comment by adibbley** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,11 +9,16 @@ source:
...
-  number: 1
-  skip: true  # [osx or win]
+  number: 2
+  noarch: generic
 
 requirements:
   run:
+    - __linux               # [linux]
+    - __win                 # [win]
+    - __archspec * x86_64   # [x86_64]
```
Comment (jakirkham) [inline recipe/meta.yaml:19]: Interesting this use to be a valid way to check `x86_64` generally

Though think some things in the stack have shifted due to recent work around `archspec`. As a result this now errors `__archspec` reports the microarchitecture. For example [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=815745&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=107 ) (also [attached log]( https://github.com/conda-forge/cuda-libraries-static-feedstock/files/13231889/cuda-libraries-static-feedstock_linux_64_20231101.5.txt ) for posterity) we can see this is `broadwell` (see below)

So we may need another approach

<details>

<summary>Result of <code>conda info</code>:</summary>

```
     active environment : base
    active env location : /opt/conda
            shell level : 1
       user config file : /home/conda/.condarc
 populated config files : /opt/conda/.condarc
                          /home/conda/.condarc
          conda version : 23.7.4
    conda-build version : 3.24.0
         python version : 3.10.13.final.0
       virtual packages : __archspec=1=broadwell
                          __glibc=2.17=0
                          __linux=6.2.0=0
                          __unix=0=0
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/conda/feedstock_root/build_artifacts/pkg_cache
                          /opt/conda/pkgs
       envs directories : /opt/conda/envs
                          /home/conda/.conda/envs
               platform : linux-64
             user-agent : conda/23.7.4 requests/2.31.0 CPython/3.10.13 Linux/6.2.0-1015-azure centos/7.9.2009 glibc/2.17
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False
```

</details>

Comment (jakirkham) [reply recipe/meta.yaml:19]: @conda-forge/core , is there another approach we should consider taking here in light of the `archspec` changes?

Comment (hmaarrfk) [reply recipe/meta.yaml:19]: i feel like `__linux` should be expanded to
```
__linux_64
__linux_ppc64le
__linux_aarch64
```

similarly for osx.

I did not know a way to better differentiate for noarch packages, so I omitted a noarch optimization due to similar constraint

Comment (jakirkham) [reply recipe/meta.yaml:19]: Opened issue ( https://github.com/conda/conda/issues/13420 ) to discuss how to capture macroarchitecture information

Comment (isuruf) [reply recipe/meta.yaml:19]: Why would you need this package to be `noarch: generic` in the first place? If you want to check `linux` and `x86_64`, simply don't make it noarch.

**Comment by adibbley** [comment]
Cancelling this, we weren't able to do this for the other cuda-libraries packages.

**Pull request closed without merging**